src/tests/ftest/launch.py

```diff
@@ -660,233 +660,162 @@ def get_test_files(test_list, args, yaml_dir, vmd_flag=False):
+        output_data = list(task.iter_buffers())
+        if len(output_data) > 1:
+            print("ERROR: Non-homogeneous {} PCI addresses.".format(device_type))
+        elif len(output_data) == 1:
+            # Get the devices from the successful, homogeneous command output
+            output_str = "\n".join([line.decode("utf-8") for line in output_data[0][0]])
+            devices = find_pci_address(output_str)
 
-    # Get the only VMD device addresses which has NVMe device connected.
-    devices = get_vmd_address_backed_nvme(host_list, all_vmd_devices)
-    print("Auto-detected VMD device which has NVMe devices connected {}: {}"
-          .format(host_list, devices))
+    return devices
 
-    return ",".join(devices), vmd_include_flag
 
-
-def get_vmd_address_backed_nvme(host_list, value):
+def get_vmd_address_backed_nvme(host_list, vmd_disks, vmd_controllers):
     """Find valid VMD address which has backing NVMe.
 
     Args:
         host_list (list): list of hosts
-        value (list): list of all PCI address.
+        vmd_disks (list): list of PCI domain numbers for each VMD controlled disk
+        vmd_controllers (list): list of PCI domain numbers for each VMD controller
 
     Returns:
-        list: a list of the VMD PCI addresses only which has connected NVMe
-              devices.
+        list: a list of the VMD controller PCI domain numbers which are connected to the VMD disks
 
     """
-    command = "ls -l /sys/block/ | grep nvme | cut -d\'>\' -f2 | cut -d'/' -f4"
-
+    disk_controllers = []
+    command_list = [
+        "ls -l /sys/block/",
+        "grep nvme",
+        "grep -E '({0})'".format("|".join(vmd_disks)),
+        "cut -d'>' -f2",
+        "cut -d'/' -f4",
+    ]
+    command = " | ".join(command_list) + " || :"
     task = get_remote_output(host_list, command)
 
     # Verify the command was successful on each server host
     if not check_remote_output(task, command):
-        print("ERROR: Issuing commands ls -l /sys/block/")
-        sys.exit(1)
-
-    # Verify each server host has the same NVMe device behind VMD addresses.
-    output_data = list(task.iter_buffers())
-    if len(output_data) > 1:
-        print("ERROR: Non-homogeneous NVMe device behind VMD addresses.")
-        sys.exit(1)
-
-    output_str = "\n".join([line.decode("utf-8") for line in output_data[0][0]])
-
-    # Remove the VMD PCI address if no NVMe is backed-up and connected.
-    for device in value:
-        if device not in output_str:
-            value.remove(device)
+        print("ERROR: Issuing command '{}'".format(command))
+    else:
+        # Verify each server host has the same NVMe device behind VMD addresses.
+        output_data = list(task.iter_buffers())
+        if len(output_data) > 1:
+            print("ERROR: Non-homogeneous NVMe device behind VMD addresses.")
+        elif len(output_data) == 1:
+            # Remove the VMD PCI address if no NVMe is backed-up and connected.
```

Comment:
this comment doesn't seem to clearly describe what the code is doing, maybe it should describe what is being added rather than what is being removed

Reply:
Thanks, I've updated the comment